gm4_ink_spitting_squid/data/gm4_ink_spitting_squid/functions/main.mcfunction

```diff
@@ -1,3 +1,3 @@
-execute as @a[gamemode=!spectator] at @s if block ~ ~1 ~ water if entity @e[type=squid,distance=..2.2] run function gm4_ink_spitting_squid:squid_squirt
+execute at @e[type=squid] as @a[gamemode=!spectator,distance=..3] at @s positioned ^ ^ ^ if block ~ ~ ~ water run function gm4_ink_spitting_squid:squid_squirt
```

Comment:
Don't you need `anchored eyes` here? Otherwise `positioned ^ ^ ^` does nothing and it will check at the player's feet.
```suggestion
execute at @e[type=squid] as @a[gamemode=!spectator,distance=..3] at @s anchored eyes positioned ^ ^ ^ if block ~ ~ ~ water run function gm4_ink_spitting_squid:squid_squirt
```